driverService/src/test/java/com/modsen/driverservice/controller/CarControllerTest.java

```diff
@@ -0,0 +1,385 @@
+
+    @Test
+    void getPageCars_withoutParams_thenReturns201() throws Exception {
+        mockMvc.perform(get(URL))
+                .andExpect(status().isOk());
+        verify(carService, times(1)).getPageCars(0,5);
+    }
+
+    @Test
+    void getPageCars_withValidParams_thenReturns201() throws Exception {
+        mockMvc.perform(get(URL).param("offset", "0").param("limit", "1"))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void getPageCars_withInvalidParams_thenReturns400() throws Exception {
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
+                List.of(offsetInvalid, limitInvalid));
+
+        MvcResult mvcResult = mockMvc.perform(get(URL).param("offset", "-1").param("limit", "-1"))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+        ListErrorMessage actualErrorResponse = objectMapper.readValue(actualResponseBody, ListErrorMessage.class);
+
+        assertThat(actualErrorResponse.errorCode()).isEqualTo(expectedErrorResponse.errorCode());
+        assertThat(actualErrorResponse.errorMessages()).containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
+    }
+
+    @Test
+    void getPageCars_withBigLimit_thenReturns400() throws Exception {
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
+                List.of(limitBig));
+
+        MvcResult mvcResult = mockMvc.perform(get(URL).param("offset", "100").param("limit", "100"))
+                .andExpect(status().isBadRequest())
+                .andReturn();
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+        ListErrorMessage actualErrorResponse = objectMapper.readValue(actualResponseBody, ListErrorMessage.class);
+
+        assertThat(actualErrorResponse.errorCode()).isEqualTo(expectedErrorResponse.errorCode());
+        assertThat(actualErrorResponse.errorMessages()).containsExactlyInAnyOrderElementsOf(expectedErrorResponse.errorMessages());
+    }
+
+    @Test
+    void getCar_whenValidId_thenReturns201() throws Exception {
+        mockMvc.perform(get(URL + "/{carId}", carId))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void getCar_whenValidInput_thenReturnsResponseDto() throws Exception {
+        when(carService.getCarById(carId)).thenReturn(carResponseDto);
+        MvcResult mvcResult = mockMvc.perform(get(URL + "/{carId}", carId))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String actualResponseBody = mvcResult.getResponse().getContentAsString();
+
+        assertThat(actualResponseBody).isEqualToIgnoringWhitespace(objectMapper.writeValueAsString(carResponseDto));
+    }
+
+    @Test
+    void getCar_whenInvalidId_thenReturns400AndErrorResult() throws Exception {
+        ListErrorMessage expectedErrorResponse = new ListErrorMessage(400,
```

Comment:
Для http статусов есть enum:
```suggestion
        ListErrorMessage expectedErrorResponse = new ListErrorMessage(HttpStatus.BAD_REQUEST.value(),
```